### PR TITLE
[QEff. Finetune] Stop fine tuning when loss has converged

### PR DESCRIPTION
**Pull request by quic-swatia**
Fine tuning will stop once the loss becomes close to convergence_loss (whose default value is <=1e-4) for few consecutive iterations, i.e., convergence_counter (by default 5).
Corrected the url of grammar dataset in finetune.md.

### CODE REVIEW

**Review by quic-mamta — APPROVED** [review]

File: QEfficient/finetune/configs/training.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ class train_config:
     intermediate_step_save: int = 1000
     batching_strategy: str = "packing"
     enable_sorting_for_ddp: bool = True
+    convergence_counter: int = 5  # stop fine training when loss is close to 0 for #convergence_counter steps
```
Comment (anujgupt-github) [inline QEfficient/finetune/configs/training.py:42]: can you avoid magic numbers in the entire PR?
Also this change will let the finetuning go on for 5 more epochs when the loss <1e-4?
what's the rationale?

Comment (quic-swatia) [reply QEfficient/finetune/configs/training.py:42]: Fine tuning will stop immediately after the loss <= 1e-4 for #convergence_counter consecutive iterations. In the attached screenshot, fine tuning stopped in between for 1st epoch because above condition was met and it didn't proceed with 2nd epoch. --num_epochs was passed as 5 for this run. 
loss-convergence

**Comment by quic-rishinr** [comment]
we would not be able to merge this feature in release 1.19 branch.

**Comment by quic-swatia** [comment]
For 1.19.3 release, the changes will get merged directly through main branch. PR for it: https://github.com/quic/efficient-transformers/pull/257
